### PR TITLE
Added cpu utilization metric & updated tests

### PR DESCRIPTION
**Pull request by james-bebbington**
**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/847

**Description:** Added cpu utilization metric & updated tests

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_metadata.go [hunk 1]
```diff
@@ -49,3 +49,13 @@ var cpuTimeDescriptor = func() pdata.MetricDescriptor {
 	descriptor.SetType(pdata.MetricTypeMonotonicDouble)
 	return descriptor
 }()
+
+var cpuUtilizationDescriptor = func() pdata.MetricDescriptor {
+	descriptor := pdata.NewMetricDescriptor()
+	descriptor.InitEmpty()
+	descriptor.SetName("system.cpu.utilization")
+	descriptor.SetDescription("CPU utilization broken down by different states.")
+	descriptor.SetUnit("%")
```
Comment (bogdandrutu) [inline receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_metadata.go:58]: Unit is "1"? I did not see "%" in the list of standard units.

File: receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper_linux.go [hunk 2]
```diff
@@ -25,12 +25,24 @@ import (
...
+	initializeCPUDataPoint(ddps.At(startIdx+1), startTime, cpuTime.CPU, systemStateLabelValue, cpuTime.System)
+	initializeCPUDataPoint(ddps.At(startIdx+2), startTime, cpuTime.CPU, idleStateLabelValue, cpuTime.Idle)
+	initializeCPUDataPoint(ddps.At(startIdx+3), startTime, cpuTime.CPU, interruptStateLabelValue, cpuTime.Irq)
+	initializeCPUDataPoint(ddps.At(startIdx+4), startTime, cpuTime.CPU, niceStateLabelValue, cpuTime.Nice)
+	initializeCPUDataPoint(ddps.At(startIdx+5), startTime, cpuTime.CPU, softIRQStateLabelValue, cpuTime.Softirq)
+	initializeCPUDataPoint(ddps.At(startIdx+6), startTime, cpuTime.CPU, stealStateLabelValue, cpuTime.Steal)
+	initializeCPUDataPoint(ddps.At(startIdx+7), startTime, cpuTime.CPU, waitStateLabelValue, cpuTime.Iowait)
+}
+
+func appendCPUUtilizationStateDataPoints(ddps pdata.DoubleDataPointSlice, startIdx int, startTime pdata.TimestampUnixNano, cpuTime cpu.TimesStat, prevCPUTime cpu.TimesStat) {
```
Comment (bogdandrutu) [inline receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper_linux.go:38]: Should utilization be computed without the state label?

File: receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_metadata.go (hunk 1 above)
Comment (james-bebbington) [reply receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_metadata.go:58]: Where is the list of standard units btw? I could not find this anywhere and had thought 1 meant a "count".

File: receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper_linux.go (hunk 2 above)
Comment (james-bebbington) [reply receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper_linux.go:38]: Yea that's a good question. I was tossing up between this myself but figured it's better to retain the state info in case people are interested (also admittedly I'm biased because Cloud Monitoring expects these labels).

It's easy to aggregate away the labels later, but not to add them back.

File: receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_metadata.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_metadata.go:58]: You can check in the proto there is a link and in the specs I believe. We also use the same definition as Stackdriver so you can check internally.

Comment (james-bebbington) [reply receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_metadata.go:58]: Cloud Monitoring, OC, & OT protos all reference http://unitsofmeasure.org/ucum.html which I found very hard to follow as it's not clear which column we should be looking at, etc.

If we are reading from the "definition unit" in that doc, then I guess 1 should be used for a [0-1] percentage value, but that seems very non-descriptive to me.

**Comment by jrcamp** [comment]
I have the same concern here about whether fine grained utilizations is the right way to go. But we can have that discussion just once in the memory PR: https://github.com/open-telemetry/opentelemetry-collector/pull/1265#issuecomment-655204709

File: receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_metadata.go (hunk 1 above)
Comment (james-bebbington) [reply receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_metadata.go:58]: Changed to [0-100] values as discussed

**Comment by james-bebbington** [comment]
See https://github.com/open-telemetry/opentelemetry-collector/pull/1265#issuecomment-656490349
